Infodev doc review (#944)
IntelPython/dpctl: docs/docfiles/user_guides/manual/dpctl/queues.rst

Queue
#####

You need a queue to schedule the execution of any computation or data copying on a
device. 

The queue construction requires specifying:

* Device
* Context targeting the device
* Additional properties, such as:
  * If profiling information should be collected
  * If submitted tasks are executed in the order, in which they are submitted

The :class:`dpctl.SyclQueue` class represents a queue and abstracts the

> :sycl_queue:`sycl::queue <>` SYCL* runtime class.

Reviewer: ```suggestion
:sycl_queue:`sycl::queue <>` SYCL\* runtime class.
```

Types of Queues
---------------

SYCL* has a task-based execution model. The order, in which a SYCL* runtime
executes a task on a target device, is specified by a sequence of events that
must be completed before the execution of the task is allowed. 

Submission of a task returns an event that you can use to further grow the graph of computational

> tasks. A SYCL* queue stores the needed data to manage the scheduling operations.

Reviewer: ```suggestion
tasks. A SYCL\* queue stores the needed data to manage the scheduling operations.
```

There are two types of queues: 

> * **Out-of-order.** Unless specified otherwise during the constriction of a queue, a SYCL* runtime

Reviewer: ```suggestion
* **Out-of-order.** Unless specified otherwise during the constriction of a queue, a SYCL\* runtime
```

  executes tasks, which dependencies are met in an unspecified order, with the
  possibility for some of the tasks to be executed concurrently.

> * **In-order.** You can specify SYCL* queues to indicate that runtime must execute tasks in the

Reviewer: ```suggestion
* **In-order.** You can specify SYC\L* queues to indicate that runtime must execute tasks in the
```

    :caption: Constructing SyclQueue from filter selector
    :linenos:

A possible output for the :ref:`fig-constructing-queue-filter-selector` example:

.. program-output:: python ../examples/python/sycl_queue.py -r create_queue_from_filter_selector

Profiling a Task Submitted to a Queue
-------------------------------------

The result of scheduling the execution of a task on a queue is an event. You can use
an event for several purposes: 

* Query for the status of the task execution
* Order execution of future tasks after it is completed
* Wait for execution to complete
* Сarry information to profile the task execution

The profiling information is only populated if the queue
used is created with the ``enable_profiling`` property and only becomes available
after the task execution is complete.

The :class:`dpctl.SyclTimer` class implements a Python context manager. 

> You can use this manager to collect cumulative profiling information for all the tasks submitted

Reviewer 2: ```suggestion
You can use this context manager to collect cumulative profiling information for all the tasks submitted
```
